ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml

```diff
@@ -0,0 +1,161 @@
+    /opt/csw/bin/pkgutil -y -i pkgutil
+    /usr/sbin/pkgchk -L CSWpkgutil
+  when: pkgutil_installed.rc != 0
+  tags: build_tools
+
+- name: Install Build Tool Packages
+  pkgutil: "name={{ item }} state=present"
+  with_items: "{{ Build_Tool_Packages }}"
+  tags: build_tools
+
+- name: Check if Solaris Studio 12.3 is installed
+  stat:
+    path: /opt/solarisstudio12.3/bin/
+  register: solaris_studio
+  tags:
+    - build_tools
+    - solaris_studio
+    - adoptopenjdk
+
+- name: Check if Solaris Studio 12.3 is downloaded
+  stat:
+    path: /tmp/SolarisStudio12.3-solaris-x86-pkg.tar.bz2
+  register: solaris_studio_downloaded
+  when: not solaris_studio.stat.exists
+  tags:
+    - build_tools
+    - solaris_studio
+    - adoptopenjdk
+
+# Stored in Azure Blob Storage (SAS URL set to expire in 2030)
+- name: Download Solaris Studio 12.3
+  # Original file is at https://www.oracle.com/tools/developerstudio/downloads/solaris-studio-v123-downloads.html
+  command: wget "https://ansiblestorageadopt.blob.core.windows.net/solarisstudiox64/SolarisStudio12.3-solaris-x86-pkg.tar.bz2?{{ SolarisStudio12_solaris_x86_SAS_TOKEN }}" -O /tmp/SolarisStudio12.3-solaris-x86-pkg.tar.bz2
+  when: not solaris_studio.stat.exists and not solaris_studio_downloaded.stat.exists
+  tags:
+    - build_tools
+    - solaris_studio
+    - adoptopenjdk
+
+- name: Extract Solaris Studio 12.3
+  shell: |
+    set -o pipefail
+    bzcat SolarisStudio12.3-solaris-x86-pkg.tar.bz2 | /bin/tar -xf -
+  args:
+    chdir: /tmp
+  tags:
+    - build_tools
+    - solaris_studio
+    - adoptopenjdk
+  when: not solaris_studio.stat.exists
+
+- name: Install Solaris Studio 12.3
+  shell: ./SolarisStudio12.3-solaris-x86-pkg/solarisstudio.sh --non-interactive
+  args:
+    chdir: /tmp
+  tags:
+    - build_tools
+    - solaris_studio
+    - adoptopenjdk
+  when: not solaris_studio.stat.exists
+
+- name: Check if cups is installed
+  stat:
+    path: /opt/sfw/cups
+  register: cups_installed
+  tags:
+    - build_tools
+    - cups
+
+- name: Download cups 1.4.4
+  command: wget https://github.com/apple/cups/releases/download/release-1.4.4/cups-1.4.4-source.tar.gz -O /tmp/cups-1.4.4-source.tar.gz
+  when: not cups_installed.stat.exists
+  tags:
+    - build_tools
+    - cups
+
+- name: Extract cups 1.4.4
+  shell: tar -xf cups-1.4.4-source.tar.gz
+  args:
+    chdir: /tmp
+  when: not cups_installed.stat.exists
+  tags:
+    - build_tools
+    - cups
+
+- name: Compile cups 1.4.4
+  shell: |
+    cd /tmp/cups-1.4.4
+    ./configure --disable-gssapi --disable-ldap --prefix=/opt/sfw/cups
+    gmake
+    gmake install
+  environment:
+    PATH: /opt/solarisstudio12.3/bin/:/opt/csw/bin/:/usr/ccs/bin:/usr/bin
+  when: not cups_installed.stat.exists
+  tags:
+    - build_tools
+    - cups
+
+- name: Create symlinks
+  file:
+    src: /opt/csw/bin/g{{ item }}
+    dest: /opt/csw/bin/{{ item }}
+    owner: root
+    group: root
+    state: link
+  with_items:
+    - tar
+    - make
+    - sha256sum
```

Comment:
Do we really need this for `sha256sum`? That would look rather odd

Reply:
yup it's a required symlink. As part of the build scripts we run sha256sum to generate the checkum

Reply:
I know `sha256sum` is used but what tries to invoke it as `gsha256sum` which seems to be what this symlink operation is creating unless I'm missing something?

Reply:
Other way round, it symlinks gsha256sum to sha256sum. Also gmake to make etc

Reply:
Eeeow - surprised it creates that one with a g prefix by default. I'm still not sure about the other two though - pretty sure we have stuff in our scripts that compensates for replacing tar and make with the g prefixed ones (search gtar and gmake in the build repo), and as per other comment I don't think it's right to switch the defaults on the system and potentially cause unecessary confusion